lib/Onyx.js

```diff
@@ -509,7 +509,9 @@ function keyChanged(key, data, canUpdateSubscriber) {
         if (_.isFunction(subscriber.callback)) {
             if (isCollectionKey(subscriber.key) && subscriber.waitForCollectionCallback) {
                 const cachedCollection = getCachedCollection(subscriber.key);
-                cachedCollection[key] = data;
+                if (data !== null) {
+                    cachedCollection[key] = data;
+                }
```

Comment:
For data `null`, don't we need to remove that key from the `cachedCollection` as we are not updating with any other value?

Reply:
I agree. Gonna ask the question another way:

Why do we need to set `data` to the `key` for the `cachedCollection`? Shouldn't the cache _already_ be correct by this point? If we are adding a key it should be there. If we are removing a key it should be there.

Reply:
Added assertions to ensure that keys that are being deleted due to being null are dropped from both cache and storage.